File: core/src/main/java/net/weweave/tubewarder/service/model/ErrorCode.java
package net.weweave.tubewarder.service.model;

public class ErrorCode {
    public static final int OK = 0;
    public static final int INVALID_INPUT_PARAMETERS = 1;
    public static final int OBJECT_LOOKUP_ERROR = 2;
    public static final int PERMISSION_DENIED = 3;
    public static final int AUTH_REQUIRED = 4;
    public static final int TEMPLATE_CORRUPT = 5;
    public static final int MISSING_MODEL_PARAMETER = 6;

    public static final int FIELD_REQUIRED = 101;
    public static final int FIELD_NAME_ALREADY_EXISTS = 102;
    public static final int FIELD_INVALID = 103;
}

File: clients/php/ErrorCode.php
<?php
namespace Tubewarder;

abstract class ErrorCode {
    const OK = 0;
    const INVALID_INPUT_PARAMETERS = 1;
    const OBJECT_LOOKUP_ERROR = 2;
    const PERMISSION_DENIED = 3;
    const AUTH_REQUIRED = 4;
    const TEMPLATE_CORRUPT = 5;
    const MISSING_MODEL_PARAMETER = 6;

    const FIELD_REQUIRED = 101;
    const FIELD_NAME_ALREADY_EXISTS = 102;
    const FIELD_INVALID = 103;
}